src/lib/dnssd/Resolver.h

```diff
@@ -439,6 +485,67 @@ class Resolver
      */
     virtual CHIP_ERROR ReconfirmRecord(const char * hostname, Inet::IPAddress address, Inet::InterfaceId interfaceId) = 0;
 
+    /**
+     * @brief Start Browsing for operational devices.
+     *
+     * This will start periodical browsing of devices. When a new operational node is
+     * found the delegate set with SetBrowseDelegate is called. The mDNS queryies
+     * will continue until StopBrowse is called.
+     */
+    virtual CHIP_ERROR StartBrowseOperationals(Optional<uint64_t> compressedFabricIdFilter) = 0;
+
+    /**
+     * @brief Start Browsing for commissionable devices.
+     *
+     * This will start periodical browsing of devices. When a new commissionable node is
+     * found the delegate set with SetBrowseDelegate is called. The mDNS queryies
+     * will continue until StopBrowse is called.
+     */
+    virtual CHIP_ERROR StartBrowseCommissionables() = 0;
+
+    /**
+     * @brief Stop browsing for devices.
+     */
+    virtual CHIP_ERROR StopBrowse() = 0;
```

Comment:
There are multiple start browsing, it seems inconvenient to have a global stop of browsing. Why don't we have APIs that can control what we stop individually?

Reply:
I have updated the code to only use 'StartBrowse'. The methods has 2 signatures, one with an optional `compressedFabricId` and one without. I agree that ideally it would be nice to have a way to individually stop a browse.
But at the moment I have not seen a use case for doing both commissionable and operational browse from the same resolver instance so I have not implemented it.